ext/tokenizer/tokenizer.c

```diff
@@ -392,6 +392,8 @@ static zend_bool tokenize(zval *return_value, zend_string *source, zend_class_en
 	array_init(return_value);
 
 	while ((token_type = lex_scan(&token, NULL))) {
+		if (token_type == T_ERROR) break;
```

Comment:
Does Tokenizer really want to return `T_ERROR` tokens to the caller? I doubt it. But if so, this can be fixed up in a different way.

Reply:
The contract here is that the lexer should only return `T_ERROR` if `PARSER_MODE()` is enabled. (We should probably `ZEND_ASSERT(token_type != T_ERROR)` here though.)

Reply:
Wow, thanks for that information!! Just looked at `PARSER_MODE()` and it seems that is is true if the tokens are actually being used to build an AST. Hope that is right?

I thought that I *had* to return `T_ERROR` because of `zend_compile.c:1810`, which asserts that the returned token must be `T_ERROR` if an exception is thrown.

Can I try to fix that?